src/main/java/com/zendesk/maxwell/schema/columndef/DateTimeColumnDef.java

```diff
@@ -20,29 +39,71 @@ private static SimpleDateFormat getDateTimeFormatter() {
 		return dateTimeFormatter;
 	}
 
+	public static int convertNanos(int nanos, Long columnLength, boolean convertStringNanosToNanos) {
+		String strNanos = Integer.toString(nanos);
+		StringBuilder realStrNanos = threadLocalBuilder.get();
+		realStrNanos.append(strNanos);
+
+		// When we have in a datetime .1234 it's actually 123400000 nano secs,
+		// not 1243
+		if ( convertStringNanosToNanos ) {
+			int i = 0;
+			while (i < 9 - strNanos.length()) {
+				realStrNanos.append('0');
+				i++;
+			}
+		}
+
+		nanos = Integer.parseInt(realStrNanos.toString());
+		int micros = nanos / 1000;
+		String strMicros = Integer.toString(micros);
+
+    // 6 is the max precision of datetime2 in MysQL
+		int divideBy = ((int) Math.pow(10, 6 - columnLength));
+
+		return micros / divideBy;
```

Comment:
I renamed it to `truncateNanosToColumnLength`. For example if you have as input `12345` and your column length is `3` you'll have `123`. `1` for a column length of `3` will give `0`.